### PR TITLE
drop python<=3.7 support

### PR DESCRIPTION
**Pull request by kloczek**
According to https://endoflife.date/python python 3.7 has been EOSed almost year ago.
Filer all code over `pyup[grade --py38-plus`.

### CODE REVIEW

**Comment by mike-fabian** [comment]
Unfortunately openSUSE Leap 15.4 and Leap 15.5 still have Python 3.6:

https://news.opensuse.org/2023/06/06/leap-modern-python-stack/

I introduced this to make it work on these openSUSE versions:

```
if sys.version_info >= (3, 8):
    from typing import Literal
else:
    from typing_extensions import Literal
from typing import Literal
```

Without that, one gets

```
ImportError: cannot import name 'Literal'
```

on openSUSE Leap 15.4 and Leap 15.5.

**Comment by mike-fabian** [comment]
Your second patch "filter all code over ruff" looks completely fine.

File: engine/emoji_picker.py
Comment (mike-fabian) [inline engine/emoji_picker.py:1]: OK

**Comment by kloczek** [comment]
> Without that, one gets
> 
> ```
> ImportError: cannot import name 'Literal'
> ```
> 
> on openSUSE Leap 15.4 and Leap 15.5.

`typing` is standard module which have been evolving since python 3..5.

Which one version of the python is in that distro? 🤔

**Comment by kloczek** [comment]
From https://peps.python.org/pep-0586/
image
So I'm assuming that this SuSE version is using one od the oleder versions which already is EOSed 🤔

**Comment by mike-fabian** [comment]
> > Without that, one gets
> > ```
> > ImportError: cannot import name 'Literal'
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > on openSUSE Leap 15.4 and Leap 15.5.
> 
> `typing` is standard module which have been evolving since python 3..5.
> 
> Which one version of the python is in that distro? 🤔

openSUSE Leap 15.5 and 15.4 have Python 3.6 as default. There are optional addons for newer Python versions, but they lack some modules. 

`typing` is available since python 3.5, but `Literal` needs python >= 3.8.

**Comment by mike-fabian** [comment]
> Your second patch "filter all code over ruff" looks completely fine.

I pushed that one to `main` already.

File: engine/factory.py
Comment (mike-fabian) [inline engine/factory.py:1]: OK

File: engine/hunspell_suggest.py
Comment (mike-fabian) [inline engine/hunspell_suggest.py:1]: OK

File: engine/hunspell_table.py
Comment (mike-fabian) [inline engine/hunspell_table.py:1]: OK for Python >= 3.6.

File: engine/itb_active_window.py
Comment (mike-fabian) [inline engine/itb_active_window.py:1]: OK

File: engine/itb_emoji.py
Comment (mike-fabian) [inline engine/itb_emoji.py:1]: OK for Python >= 3.6

**Comment by kloczek** [comment]
https://github.com/mike-fabian/ibus-typing-booster/blob/91c9296341a612f65b5a514f097279d9a6ed05d5/configure.ac#L42-L43
It would be good to update that as well

File: engine/itb_nltk.py
Comment (mike-fabian) [inline engine/itb_nltk.py:1]: OK for python >= 3.6.

File: engine/itb_pango.py
Comment (mike-fabian) [inline engine/itb_pango.py:1]: OK

File: engine/itb_sound.py
Comment (mike-fabian) [inline engine/itb_sound.py:1]: OK

**Comment by mike-fabian** [comment]
> https://github.com/mike-fabian/ibus-typing-booster/blob/91c9296341a612f65b5a514f097279d9a6ed05d5/configure.ac#L42-L43
> 
> It would be good to update that as well

Yes, but not  right now because I think I stlll need to keep Python 3.6 compatibility because of openSUSE Leap 15.5.

**Comment by mike-fabian** [comment]
I have pushed all changes not to main **except** for the  change related to `from typing import Literal`. That keeps Python 3.6 compatibility for the moment.

**Comment by mike-fabian** [comment]
Thank you for the pull request!

**Pull request closed without merging**